ui/components/app/snaps/snap-ui-renderer/snap-ui-renderer.js

```diff
@@ -72,44 +196,47 @@ const SnapUIRendererComponent = ({
 
   const backgroundColor =
     contentBackgroundColor ??
-    mapToExtensionCompatibleColor(content?.props?.backgroundColor) ??
+    mapToExtensionCompatibleColor(processedContent?.props?.backgroundColor) ??
     BackgroundColor.backgroundAlternative;
 
+  // The renderer should only have a footer if there is a default cancel action
+  // or if the footer component has been used.
+  const hasFooter =
+    onCancel || processedContent?.props?.children?.[1] !== undefined;
+
   const sections = useMemo(
     () =>
-      content &&
+      processedContent &&
       mapToTemplate({
         map: {},
-        element: content,
+        element: processedContent,
         onCancel,
         useFooter,
         promptLegacyProps,
         t,
         contentBackgroundColor: backgroundColor,
+        // We have to pass the props at the top level to ensure the children components get the correct props.
+        requireScroll,
+        isScrolledToBottom: scrollState.buttonsEnabled,
       }),
-    [content, onCancel, useFooter, promptLegacyProps, t, backgroundColor],
+    [
+      processedContent,
```

Comment:
If I am understanding this correctly we are remapping the components every time you scroll. This won't work for performance reasons.

Perhaps we can explore exposing the scroll state via the interface context instead of remapping and rerendering everything.